tests/basic.rs

```diff
@@ -393,10 +393,9 @@ fn different_size_maps_not_equal() {
     let map1 = HashMap::<usize, usize>::new();
     let map2 = HashMap::<usize, usize>::new();
     {
-        let guard = map1.guard();
-        map1.insert(1, 0, &guard);
-        map1.insert(2, 0, &guard);
-        map2.insert(1, 0, &guard);
+        map1.pin().insert(1, 0);
```

Comment:
Why these changes? This changes what the test actually tests a bit, since it no longer re-uses a single guard.

Reply:
`map1` and `map2` have to use different guards, because they aren't tied to the same collector (which used to be the global one). It could do `let guard1 = map1.guard(); let guard2 = map2.guard();` though.

Reply:
Yeah, I was more thinking about the repeated `map.pin()`. Having two guards that each get reused for their respective maps seems fine.

Reply:
Fixed in d50737c467f47b6732057d36d29fbaa9a0751765.